clippy_lints/src/unnecessary_wraps.rs

```diff
@@ -159,15 +159,22 @@ impl<'tcx> LateLintPass<'tcx> for UnnecessaryWraps {
                 )
             };
 
-            span_lint_and_then(cx, UNNECESSARY_WRAPS, span, lint_msg, |diag| {
-                diag.span_suggestion(
-                    fn_decl.output.span(),
-                    return_type_sugg_msg,
-                    return_type_sugg,
-                    Applicability::MaybeIncorrect,
-                );
-                diag.multipart_suggestion(body_sugg_msg, suggs, Applicability::MaybeIncorrect);
-            });
+            span_lint_and_then(
+                cx,
+                UNNECESSARY_WRAPS,
+                // Do not include the whole body in the lint span
+                trim_span(cx.sess().source_map(), span.until(body.value.span)),
```

Comment:
I believe this could be `cx.tcx.def_span(def_id)`

Reply:
Neat, I didn't know this one. Updated.